rfcs/require-by-string.md

```diff
@@ -0,0 +1,360 @@
+
+- Libraries are fully compatible with the Roblox engine, as require-by-string is currently unimplemented.
+- Luau currently implements relative paths in relation to the current working directory. 
+- We propose changing this behavior and breaking backwards compatibility on this front.
+- With the current implementation, requiring a library that itself contains relative-path require statements [can become a mess](https://github.com/Roblox/luau/issues/959) if the Luau VM is not launched from the "correct" working directory.
+- We propose the following change: relative paths passed to require statements will be evaluated in relation to the requiring file's location, not in relation to the current working directory.
+    - Caveat: if the relative-path require is called directly in the Luau CLI, this will remain relative to the current working directory, as there is no sense of a "requiring file" here.
+- If this causes issues, we can later introduce a default alias for the current working directory (e.g. `@CWD`).
+
+## Alternatives
+
+### Prohibiting absolute paths in require
+
+Again, using aliases to refer to absolute paths is encouraged over directly passing absolute paths to require. To promote this, we might consider prohibiting the passing of absolute paths to require statements altogether.
+
+There are two main reasons for *not* doing this.
+
+#### Fast prototyping and testing
+
+First, aliases certainly make code simple and readable, but we still want to allow developers to prototype and test their code quickly. Requiring developers to set up/update their alias map each time they want to refer to a package by its absolute path isn't ideal.
+
+If absolute paths are prohibited in require statements altogether, developers might lean toward using relative paths, which often take more effort to understand:
+```lua
+-- This is long, but the directory structure is clear
+local result = require("Users/JohnDoe/Luau/LuauModules/HelperModules/graphing")
+
+-- This is shorter, but it is unclear which directory each "../" refers to
+local result = require("../../../../../HelperModules/graphing")
+```
+
+#### Backwards compatibility
+
+Second, backwards compatibility. Luau currently uses `fopen`/`_wfopen` to locate required modules, which both support absolute paths. Any existing code that relies on absolute paths being passed to require statements will break if this behavior is changed.
+
+### Different ways of defining aliases
+
+#### Defining aliases directly in the requiring file
+
+Rather than defining an alias map in an external configuration file, we could alternatively define aliases directly in the files that require them. For example, this could manifest itself through an extension of the `--!` comment syntax or introduce new syntax like `--@<ALIAS> = @<PATH>`.
```

Comment:
I think we explicitly want aliases to be external, because they are a tool for retargeting modules.